tools/docker/docker-compose.yml

```diff
@@ -14,12 +14,27 @@ services:
             MYSQL_DATABASE: ragnarok
             MYSQL_USER: ragnarok
             MYSQL_PASSWORD: ragnarok
-    server:
+
+    builder:
```

Comment:
What does `builder` do; is it just to check that the image starts?

Reply:
Or was it to build the project for the other images but missing its `command:`

Reply:
When it's the first time you use rAthena you won't be able to shell into any other image to build the project so I created this image to just be there in case you need to build...

Do you know of any better way of tackling this issue?

Reply:
Also I didn't want to have the command there to build because then we would build everytime we start the containers

Reply:
A couple steps to have the builder build when we want it to, have a command so we wont get the daemon pid 1 nonsense 

Add a file `builder.sh` in folder `tools/docker`, run `chmod 755 builder.sh` then make its body:
```

if [ ! -f /rathena/login-server ]; then
  export runBuild=1;
elif [ ! -f /rathena/char-server ]; then
  export runBuild=1;
elif [ ! -f /rathena/map-server ]; then
  export runBuild=1;
else
  export runBuild=0;
fi

if [ "${runBuild}" -eq "1" ]; then
  ### checking that ./configure has ran by looking for make file
  if [! -f /rathena/make ]; then
    echo "Error : ./configure has not been ran successfully. Please start the docker image alone and run the command /rathena/configure on that image.";
    echo "Resolution : run the following command \"docker-compose run builder /rathena/configure\"";
    exit 1;
  fi

  make clean server;
fi
```

make builder in the docker-compose file have a `command:` line of `command: "/rathena/tools/docker/builder.sh"`
Making lines 18-21 resemble the following:
```
    builder:
        image: "rathena:local"
        container_name: "builder"
        command: "/rathena/tools/docker/builder.sh"
        ...
```        

----

Should probably also update the readme for docker to let the others know how to configure it

Namely, with updating the ```### How to setup a local development environment :computer:```'s line 
```
2. `docker exec -it rathena bash` to connect to dev container
```

to 
```
1. `docker-compose run builder bash` to connect to a dev container in order to run aditional build or shell scripts within the linux context.
```

Maybe also with some extra lines like,
```
1. All commands expect you to have change directory to this directory, `$projectRoot/tools/docker`.
1. Run the configure to setup your make files for the docker _alpine_ image `docker-compose run builder /rathena/configure`
1. If you want the builder to build your project on each start change line 8 from
\```bash
  export runBuild=0;
\```
to
\```bash
  export runBuild=1;
\```

```

and

Reply:
Also I noticed the the -c's on the other three rathena images in the cluster that the `-c` was causing issues on both a Windows 10 and a Mac-Intel - Big Sur and had to remove them, have not tried on the debian image i have yet

Reply:
Odd thing is I'm using Ubuntu as my Windows11 subsystem and couldn't get it running without the -C.

Also I've made some changes to the build.sh, instead of asking to run configure beforehand, I've added a env variable to the compose file and we always run the configure if we don't have the make file. What do you think @alycecil ?

Reply:
Sounds great!